CON-328  Add HTTP call to the slack when a meeting is ended

Ramadhan0 opened the pull request:
#### Description
This feature ensures that an HTTP call is made to the slack integration micro-service when a meeting is ended, either by the user or when the meeting time runs out.

It sends out different arguments including the organizer_email, room_id, and the event_id.

How to test it: 
- update your .env file with this variable
   `SLACK_NOTIFICATION_URL=https://www.softwaretestinghelp.com/tools/30-top-website-link-verification-testing-tools/
`
- make sure all services are up including `redis`
- open insomnia or postman and hit the end event mutation structured as below,
`mutation{
   endEvent(calendarId:"andela.com_38393338303730373538@resource.calendar.google.com", eventId:"0manph5i3e8lnld6tnvb0o1pcc",
    organizerEmail:"converge.notification@andela.com",
    roomId: "18",
startTime:"2018-08-19T23:00:00-07:00",
        endTime:"2018-08-20T05:00:00-07:00",
meetingEndTime: "2018-08-14T07:00:00-07:00"){
               event{
                eventId
                  eventTitle
                checkedIn
                cancelled
                meetingEndTime
                room{
                    id
                    name
                    calendarId
                }
            }
        }
    }`
- check-in the celery.err.log file for the response to be `200` after finishing the task with a notify-slack name.
#### Type of change
Please select the relevant option
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Others (cosmetics, styling, improvements)
- [ ] This change requires a documentation update

#### How Has This Been Tested?

- [x] Unit
- [ ] Integration
- [ ] End-to-end

New feature
#### JIRA
[CON-328](https://andela-apprenticeship.atlassian.net/browse/CON-328)
[CV3-63](https://andela-apprenticeship.atlassian.net/jira/software/projects/CV2/boards/22?selectedIssue=CV3-62)

Ramadhan0 committed:
CON-328-story: allow HTTP call to the slack when a metting is ended

MCFrank16 commented on helpers/event/event_slack_notifier.py:
```diff
@@ -0,0 +1,12 @@
+import requests
+import os
```
where will you use this os on?

MCFrank16 reviewed (approved):
Good job @Ramadhan0

joshuaocero commented on config.py:
```diff
@@ -19,6 +19,9 @@ class Config:
     CELERY_RESULT_BACKEND = os.getenv('CELERY_RESULT_BACKEND')
 
+    # slack bot url
+    notify_url = os.getenv('SLACK_NOTIFICATION_URL')
```
Please maintain consistency in variable naming. As you can see, all the other config variables are uppercased.

joshuaocero commented on helpers/event/event_slack_notifier.py:
```diff
@@ -0,0 +1,11 @@
+"""
+
+@celery.task(name="notify-slack")
+def notify(event_Id):
```
Please refer to variable naming conventions for `event_Id`. (https://www.python.org/dev/peps/pep-0008/#function-and-variable-names)

joshuaocero replied:
Please make this function name more descriptive. `notify` what? Think of a better function name. Ask yourself what exactly the function is doing then name it appropriately.

joshuaocero commented on helpers/event/event_slack_notifier.py:
```diff
@@ -0,0 +1,11 @@
+
+@celery.task(name="notify-slack")
+def notify(event_Id):
+    requests.get(url=Config.notify_url+str(event_Id))
```
Why are you casting `event_Id` to a string. Is it necessary? Also, please refer to variable naming conventions for `event_Id`. (https://www.python.org/dev/peps/pep-0008/#function-and-variable-names)

Ramadhan0 replied:
the `event_id` is an integer and the `URL` is a string
so, without converting `event_id` to a string was causing an `error` @joshuaocero

joshuaocero commented on api/events/schema.py:
```diff
@@ -96,16 +97,18 @@ def mutate(self, info, **kwargs):
+            event_reject_reason
+        ):
             raise GraphQLError("Event cancelled but email not sent")
+        event_Id=kwargs['event_id']
```
Please refer to variable naming conventions for `event_Id`. (https://www.python.org/dev/peps/pep-0008/#function-and-variable-names)

joshuaocero commented on api/events/schema.py:
```diff
@@ -96,16 +97,18 @@ def mutate(self, info, **kwargs):
+        ):
             raise GraphQLError("Event cancelled but email not sent")
+        event_Id=kwargs['event_id']
+        notify.delay(event_Id)
```
Please refer to variable naming conventions for `event_Id`. (https://www.python.org/dev/peps/pep-0008/#function-and-variable-names)

joshuaocero commented on helpers/event/event_slack_notifier.py:
```diff
@@ -0,0 +1,11 @@
+import requests
```
Make this filename leaner. It is already in a folder called `events`, no need to prefix the filename with `event_`.

joshuaocero commented on tests/test_events/test_slack_notifier.py:
```diff
@@ -0,0 +1,14 @@
+class test_slack_notifier(unittest.TestCase):
+
+    @patch('helpers.event.event_slack_notifier.requests.get')
+    def test_add_room(self, mock_get_request):
```
Why is this function called `test_add_room`? You need to pay more attention to detail. Avoid unnecessary copying and pasting.

joshuaocero reviewed (changes requested):
@Ramadhan0 there are a couple of things you need to work on. Please look into my feedback.

Commit:
(CON-328-story): add organizer_email and room_id
  - ensure the returned url is comprised with the organizer_email and room_id

[Delivers CON-328]

mifeille commented on alembic/versions/2ee259ae22e5_add_organizer_column_event.py:
```diff
@@ -0,0 +1,25 @@
+
+def upgrade():
+    op.add_column('events', sa.Column(
+        'organizer_email', sa.String(), nullable=True))
```
I just checked Samuel's PR and I saw that the organizer email is sent as a list since it is taking two emails, is this data type appropriate for storing it?

MCFrank16 replied:
I just wanna know from the list of 2 emails, how will I know which one is the actual organizer_email @mifeille

mifeille reviewed (commented):
Please validate organizerEmail and roomId fields.
